tx/tx.go

```diff
@@ -0,0 +1,373 @@
+	if numCreds < 2 {
+		return nil, nil, fmt.Errorf("expected tx.Creds of len 2, got %d. doesn't seem to be a multisig tx with subnet auth requirements", numCreds)
+	}
+	// signatures for output owners should be filled (all creds except last one)
+	for credIndex := range ms.Tx.Creds[:numCreds-1] {
+		cred, ok := ms.Tx.Creds[credIndex].(*secp256k1fx.Credential)
+		if !ok {
+			return nil, nil, fmt.Errorf("expected cred to be of type *secp256k1fx.Credential, got %T", ms.Tx.Creds[credIndex])
+		}
+		for i, sig := range cred.Sigs {
+			if sig == emptySig {
+				return nil, nil, fmt.Errorf("expected funding sig %d of cred %d to be filled", i, credIndex)
+			}
+		}
+	}
+	// signatures for subnet auth (last cred)
+	cred, ok := ms.Tx.Creds[numCreds-1].(*secp256k1fx.Credential)
+	if !ok {
+		return nil, nil, fmt.Errorf("expected cred to be of type *secp256k1fx.Credential, got %T", ms.Tx.Creds[1])
+	}
+	if len(cred.Sigs) != len(authSigners) {
+		return nil, nil, fmt.Errorf("expected number of cred's signatures %d to equal number of auth signers %d",
+			len(cred.Sigs),
+			len(authSigners),
+		)
+	}
+	remainingSigners := []ids.ShortID{}
+	for i, sig := range cred.Sigs {
+		if sig == emptySig {
+			remainingSigners = append(remainingSigners, authSigners[i])
+		}
+	}
+	return authSigners, remainingSigners, nil
+}
+
+// GetAuthSigners gets all subnet auth addresses that are required to sign a given tx
+//   - get subnet control keys as string slice using P-Chain API (GetOwners)
+//   - get subnet auth indices from the tx, field tx.UnsignedTx.SubnetAuth
+//   - creates the string slice of required subnet auth addresses by applying
+//     the indices to the control keys slice
+func (ms *SignTxResult) GetAuthSigners() ([]ids.ShortID, error) {
+	if ms.Undefined() {
+		return nil, ErrUndefinedTx
+	}
+	controlKeys, _, err := ms.GetSubnetOwners()
+	if err != nil {
+		return nil, err
+	}
+	unsignedTx := ms.Tx.Unsigned
+	var subnetAuth verify.Verifiable
+	switch unsignedTx := unsignedTx.(type) {
+	case *txs.RemoveSubnetValidatorTx:
+		subnetAuth = unsignedTx.SubnetAuth
+	case *txs.AddSubnetValidatorTx:
+		subnetAuth = unsignedTx.SubnetAuth
+	case *txs.CreateChainTx:
+		subnetAuth = unsignedTx.SubnetAuth
+	case *txs.TransformSubnetTx:
+		subnetAuth = unsignedTx.SubnetAuth
+	case *txs.TransferSubnetOwnershipTx:
+		subnetAuth = unsignedTx.SubnetAuth
+	case *txs.ConvertSubnetToL1Tx:
+		subnetAuth = unsignedTx.SubnetAuth
+	default:
+		return nil, fmt.Errorf("unable to GetAuthSigners due to unexpected unsigned tx type %T", unsignedTx)
+	}
+	subnetInput, ok := subnetAuth.(*secp256k1fx.Input)
+	if !ok {
+		return nil, fmt.Errorf("expected subnetAuth of type *secp256k1fx.Input, got %T", subnetAuth)
+	}
+	controlKeysLen := len(controlKeys)
+	if controlKeysLen > math.MaxUint32 {
+		return nil, fmt.Errorf("value %d out of range for uint32", controlKeysLen)
+	}
+	authSigners := []ids.ShortID{}
+	for _, sigIndex := range subnetInput.SigIndices {
```

Comment:
nit: can reduce number of operations here by checking on length of `SigIndices` before this for loop